### PR TITLE
docs: Update Gitlab integration page

### PR DESCRIPTION
**Pull request by vikram-dagger**
This commit adds an example of integrating with GitLab Runners using the Kubernetes executor.

### CODE REVIEW

File: docs/current_docs/integrations/734201-gitlab.mdx
Comment (vikram-dagger) [inline docs/current_docs/integrations/734201-gitlab.mdx:25]: I have updated the tolerations and affinity values in the OpenShift instructions to use `dagger-node`, so a user can easily match the runner configmap below with the node config. Hope that's OK @gerhard ?

I would like to also link to our Kubernetes integration page for a similar example of K8s configuration. However, the Kubernetes page uses our Helm chart with different values for tolerations and affinity: `dagger-runner` and `actions-runner`. See https://github.com/dagger/dagger/blob/main/helm/dagger/values.yaml. Does it make sense to update those values to also be `dagger-node`, so that it syncs with the configmap below?

File: docs/current_docs/integrations/248292-openshift.mdx
Comment (gerhard) [inline docs/current_docs/integrations/248292-openshift.mdx:14]: This is a much better name for it, I like it!

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (gerhard) [inline docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:5]: This should be a secret - I am assuming it is.

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (gerhard) [inline docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:8]: If possible, this should be already provisioned in the runner before the job starts. We are doing something similar in our latest GitHub Actions integration. @matipan shares that context.

This will be increasingly important for the next 3-6 months as the CLI & Engine architecture is getting overhauled. FTR:
- https://github.com/dagger/dagger/issues/5583
- https://github.com/dagger/dagger/issues/5484

File: docs/current_docs/integrations/snippets/openshift.yaml
Comment (gerhard) [inline docs/current_docs/integrations/snippets/openshift.yaml:7]: If this is the Dagger version, should we bump it to the recently released `v0.11.0`?

**Review by gerhard — COMMENTED** [review]
Will appreciate your input on this @chrira 👍

File: docs/current_docs/integrations/snippets/gitlab-runner-config.yml
Comment (vikram-dagger) [inline docs/current_docs/integrations/snippets/gitlab-runner-config.yml:9]:
```suggestion
      environment = ["HOME=/tmp","FF_GITLAB_REGISTRY_HELPER_IMAGE=1", "_EXPERIMENTAL_DAGGER_RUNNER_HOST"="unix:///var/run/dagger/buildkitd.sock"]
```

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (vikram-dagger) [inline docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:4]:
```suggestion
```

Comment (vikram-dagger) [reply docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:4]: I have added this change as a suggestion. Please check and confirm if OK @gerhard @chrira

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (vikram-dagger) [inline docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:5]:
```suggestion
    # assumes the Dagger Cloud token is
    # in a masked/protected variable named DAGGER_CLOUD_TOKEN
    # set via the GitLab UI
    DAGGER_CLOUD_TOKEN: $DAGGER_CLOUD_TOKEN
```

Comment (vikram-dagger) [reply docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:5]: Added a suggestion for this. Please review @gerhard @chrira

File: docs/current_docs/integrations/snippets/openshift.yaml
Comment (vikram-dagger) [inline docs/current_docs/integrations/snippets/openshift.yaml:7]:
```suggestion
    tag: v0.11.0
```

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (vikram-dagger) [inline docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:8]:
```suggestion
    - curl -sfL https://releases.dagger.io/dagger/install.sh | DAGGER_VERSION=0.11.0 BIN_DIR=/tmp sh
```

File: docs/current_docs/integrations/snippets/openshift.yaml
Comment (vikram-dagger) [reply docs/current_docs/integrations/snippets/openshift.yaml:7]: Added as suggestion

**Review by vikram-dagger — COMMENTED** [review]
@gerhard @chrira I have added your feedback as inline suggestions in this PR. Please review and confirm if OK, or add further feedback. 

I added them as inline suggestions instead of directly updating the files in the PR so the diff is more visible in context of the comments made.

**Review by chrira — COMMENTED** [review]
@vikram-dagger I did not test the changes but they look good for me 👍
Thanks!

File: docs/current_docs/integrations/734201-gitlab.mdx
Comment (vikram-dagger) [reply docs/current_docs/integrations/734201-gitlab.mdx:25]: @gerhard for above questions

**Comment by vikram-dagger** [comment]
@gerhard based on @chrira lgtm, I went ahead and committed the suggestions. Now waiting for your review to proceed.

File: docs/current_docs/integrations/snippets/openshift.yaml
Comment (vikram-dagger) [inline docs/current_docs/integrations/snippets/openshift.yaml:7]: Can we change this to `latest`? It would simplify maintenance of this page but would it work? @chrira @gerhard ?

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (vikram-dagger) [inline docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:10]: Can we change this to `latest`? It would simplify maintenance of this page but would it work? @chrira @gerhard ?

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (chrira) [reply docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:7]: LGTM

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (vikram-dagger) [reply docs/current_docs/integrations/snippets/gitlab-kubernetes.yml]: Done

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (gerhard) [inline docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:10]: Using an explicit `DAGGER_VERSION` variable here which matches the Engine version is highly encouraged. If these two diverge, it may lead to incompatibility issues between the CLI & the Engine.

`/tmp` is usually not executable, so installing the `dagger` CLI there may result in this not working, even if `/tmp` is added to `$PATH`. Regardless where `dagger` CLI gets installed, this path should either:
1. Be added to the `$PATH` explicitly
2. Invoked via an absolute path, e.g. `~/.local/bin/dagger` or `~/bin/dagger`

File: docs/current_docs/integrations/snippets/gitlab-runner-config.yml
Comment (gerhard) [inline docs/current_docs/integrations/snippets/gitlab-runner-config.yml:9]: Is setting `HOME=/tmp` correct here? I genuinely don't know what the expectation is in a GitLab Runners, but this seems odd.

File: docs/current_docs/integrations/snippets/gitlab-runner-config.yml
Comment (gerhard) [inline docs/current_docs/integrations/snippets/gitlab-runner-config.yml:30]: Double-checking that these are the correct values in GitLab Runner. The expectation is that the system user which runs the `dagger` CLI in the context of the GitLab Runner belongs to the same system group so that it is able to read Dagger's unix socket.

File: docs/current_docs/integrations/snippets/openshift.yaml
Comment (gerhard) [inline docs/current_docs/integrations/snippets/openshift.yaml:7]: Are we sure that we want `latest`? I'm not hinting that it's wrong, just double-checking that we want to run with whatever is latest when this gets executed.

Benefits of `latest`: auto-updates, less maintenance overhead.
Drawbacks of `latest`: requires explicit update, pipelines may stop working due to breaking changes (e.g. 0.8.0, 0.10.0, etc.)

Benefits of a fixed version: always know what is running in CI, pipelines won't break unless release notes are ignored.
Drawbacks of a fixed version: more maintenance, may fall behind if this is not done regularly.

Either approached are valid. OK to keep as is if it's what we want.

**Review by gerhard — APPROVED** [review]
Thank you @vikram-dagger for all the nudges! I was finally able to do one last pass. OK to merge after you read my comments. OK to skip anything which you don't feel is relevant / helpful.

Going to answer your other questions next.

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (gerhard) [reply docs/current_docs/integrations/snippets/gitlab-kubernetes.yml]: OK to keep as is - `latest` is fine for the purpose of this documentation. +1 to what @chrira wrote further below: not good for production, OK for documentation that does not go outdated & requires maintenance.

If you think that capturing the specific version approach in a comment is worth doing @vikram-dagger, that works!

File: docs/current_docs/integrations/snippets/openshift.yaml
Comment (gerhard) [reply docs/current_docs/integrations/snippets/openshift.yaml]: OK.

File: docs/current_docs/integrations/734201-gitlab.mdx
Comment (gerhard) [reply docs/current_docs/integrations/734201-gitlab.mdx:25]: 1. Yes.
2. I like it! Good change 👍

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (vikram-dagger) [reply docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:13]: - Will `~/.local` and `~/.bin` exist?
- If we change the path here, should we also change it in the `$HOME` variable in the runner config?

File: docs/current_docs/integrations/snippets/gitlab-runner-config.yml
Comment (vikram-dagger) [reply docs/current_docs/integrations/snippets/gitlab-runner-config.yml:9]: @chrira any thoughts on this?

File: docs/current_docs/integrations/snippets/gitlab-runner-config.yml
Comment (vikram-dagger) [reply docs/current_docs/integrations/snippets/gitlab-runner-config.yml:30]: @chrira any thoughts on this?

File: docs/current_docs/integrations/snippets/openshift.yaml
Comment (vikram-dagger) [reply docs/current_docs/integrations/snippets/openshift.yaml:7]: Added this info to the page

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (chrira) [reply docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:13]: We don't know wich image with wich operating system is used. Therefore we have to take a common folder, like `/tmp`.
I would also set it in the runner config. When I tested it I found following solution:
```
[[runners]]
  environment = ["HOME=/tmp", ...
  pre_build_script = "export PATH=\"/tmp/:$PATH\""
```
Setting it inside `environment` does not work. The PATH is not available at this time.

File: docs/current_docs/integrations/snippets/gitlab-runner-config.yml
Comment (chrira) [reply docs/current_docs/integrations/snippets/gitlab-runner-config.yml:9]: We need a writable home dir inside the build container.
The pod that is started by the runner can contain any Image.
As mentioned before, we do not know how the Image is built.
We use this config at Puzzle. I would leave it like this.

File: docs/current_docs/integrations/snippets/gitlab-runner-config.yml
Comment (chrira) [reply docs/current_docs/integrations/snippets/gitlab-runner-config.yml:30]: Changing those settings did not have any impact on my setup.
I would delete this two lines.

Comment (vikram-dagger) [reply docs/current_docs/integrations/snippets/gitlab-runner-config.yml:30]: Deleted

File: docs/current_docs/integrations/snippets/gitlab-runner-config.yml
Comment (gerhard) [reply docs/current_docs/integrations/snippets/gitlab-runner-config.yml:9]: OK, leaving this as is since you are already using it & it's known to work.

File: docs/current_docs/integrations/snippets/gitlab-kubernetes.yml
Comment (gerhard) [reply docs/current_docs/integrations/snippets/gitlab-kubernetes.yml:13]: Done!

**Pull request merged**
